packages/data-models/src/definition.js

```diff
@@ -1,22 +1,90 @@
-import { v4 as uuidv4 } from 'uuid'
-import ResourceProvider from './resource_provider.js'
+/** @module definition */
+import ResourceProvider from '@dmodels/resource_provider.js'
+import Language from '@dmodels/language.js' /* @typedef {import('./language.js').Language} Language */
+import IRIProvider from '@dmodels/iri/iri-provider.js'
 
 class Definition {
+  /**
+   * @param {string} text - A text of a definition.
+   * @param {Language} language - A _language on what a text of a definition is written.
+   * @param {string} format - A MIME type of a definitions text (i.e. "text/plain").
+   * @param {string} lemmaText - A ward that the definition text describes.
+   */
   constructor (text, language, format, lemmaText) {
-    this.text = text
-    this.language = language
+    if (!(language instanceof Language)) {
+      throw new Error('The language argument is not of the Language type')
+    }
+
+    /**
+     * A text of a definition.
+     *
+     * @type {string}
+     */
+    this._text = text
+
+    /**
+     * A language on what a text of a definition is written.
+     *
+     * @type {Language}
+     */
+    this._language = language
+
+    /**
+     * A MIME type of a definitions text (e.g. "text/plain").
+     *
+     * @type {string}
+     */
     this.format = format
+
+    /**
+     * A ward that the definition text describes.
+     *
+     * @type {string}
+     */
     this.lemmaText = lemmaText
 
-    this.ID = uuidv4()
+    /**
+     * A provider of the definition. Will be set outside of the constructor,
+     * currently, in the `getProxy` method of the ResourceProvider.
+     *
+     * @type {ResourceProvider}
+     */
+    this.provider = null
+
+    /**
+     * A unique ID of an object instance.
+     *
+     * @type {string}
+     */
+    this._ID = IRIProvider.getIRI({ identityData: this.identityData })
+  }
+
+  get text () {
+    return this._text
+  }
+
+  get language () {
+    return this._language
+  }
+
+  get ID () {
+    return this._ID
+  }
+
+  get identityData () {
+    return {
+      text: this._text,
+      languageCode: this._language.toCode()
```

Comment:
I think we should also include the resource provider uri in the identity data for the definition to start. There may be additional changes to the way we calculate the IRI for definitions down the road, but that would be a reasonable place to start.

Reply:
It would be fixed in a separate PR because the complicated nature of the solution would make this PR too heavy. I've created an issue to track the progress: https://github.com/alpheios-project/alpheios-core/issues/595.